ArduPlane/mode_takeoff.cpp

```diff
@@ -152,6 +149,13 @@ void ModeTakeoff::update()
         plane.calc_nav_pitch();
         plane.calc_throttle();
         //check if in long failsafe, if it is recall long failsafe now to get fs action via events call
+#if AP_FENCE_ENABLED
+        const uint16_t altitude = plane.relative_ground_altitude(false,true);
+        if (altitude >= alt && !have_auto_enabled_fence) {
```

Comment:
I think the standard loiter check can be used?
```suggestion
        if (plane.reached_loiter_target() && (loiter.reached_target_alt || loiter.unable_to_acheive_target_alt) && !have_auto_enabled_fence) {
```

Reply:
I will try it in SITL